add a new spark parameter:spark.neo4j.bolt.encryption.status

echohlne opened the pull request:
In NEO4J, if the parameter dbms.connector.bolt. Tls_level is set to REQUIRED, when we connect NEO4J with neo4j-spark connector, the connection will report an error:Connection to the database terminated. This can happen due to network instabilities, or due to restarts of the database.So We add a parameter to control how the Config instance can be created,If the encryptionStatus variable is false, use the withoutEncryption() method to initialize the config instance,otherwise use the withEncryption() to initialize it.

echohlne commented:
Does anyone know what caused check error above? Thanks a lot. 


details:

travis_fold:start:install_jdk
[0K[33;1mInstalling oraclejdk8[0m
$ export JAVA_HOME=~/oraclejdk8
$ export PATH="$JAVA_HOME/bin:$PATH"
$ ~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"
Ignoring license option: BCL -- using GPLv2+CE by default
install-jdk.sh 2019-10-16
Expected feature release number in range of 9 to 14, but got: 8
[31;1mThe command "~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"" failed and exited with 3 during .[0m

conker84 commented:
@echohlne thanks a lot for the contribution, can you please rebase the code and add add an example into the `README.md`?

echohlne commented:
@conker84 Thanks a lot for your great work in this project. I really appreciate your advice. According to your opinion, I had modified the code, so that it looks clearer, at the same time, I also updated the readme.md. Looking forward to your reply.

conker84 commented:
@echohlne sorry for the delay, did you signed the CLA? https://neo4j.com/developer/cla
Thanks a lot!

echohlne commented:
@conker84 Thank you for reminding me.  I just signed the CLA.
